pandas/tests/copy_view/test_indexing.py

```diff
@@ -892,10 +906,14 @@ def test_column_as_series(backend, using_copy_on_write, using_array_manager):
     if using_copy_on_write or using_array_manager:
         s[0] = 0
     else:
-        warn = SettingWithCopyWarning if dtype_backend == "numpy" else None
-        with pd.option_context("chained_assignment", "warn"):
-            with tm.assert_produces_warning(warn):
+        if warn_copy_on_write:
+            with tm.assert_produces_warning(FutureWarning):
                 s[0] = 0
```

Comment:
How important do we find it that we ensure the exact message? (given that this is only temporary, maybe not super important?)

(although I suppose it is not difficult to just add `match="Setting a value on a view"` to all cases I am adding in this PR.

I don't know how useful it will be to add a custom `tm.assert_produces_cow_warning`, but it would make the `tm.assert_produces_warning(FutureWarning, match="Setting a value on a view"):` a bit shorter. 

But when adding warnings for more cases, the exact warning message might also differ.

Reply:
I think it's okay being lax with matching the warning message during testing (but noting a comment that it's CoW related if it's not obvious from the test)

Reply:
> but noting a comment that it's CoW related if it's not obvious from the test

That might be a reason to actually make something like a `tm.assert_produces_cow_warning`, then it's also clear without adding a comment everywhere, and will be easier to remove all of those.